colcon_bundle/verb/_archive_generators.py

```diff
@@ -73,3 +61,88 @@ def generate_archive(install_base,
     os.remove(bundle_tar_path)
 
     logger.info('Archiving complete')
+
+
+def generate_archive_v2(install_base,
+                        staging_path,
+                        bundle_base,
+                        metadata_paths,
+                        dependencies_changed):
+    """
+    Generate bundle archive v2.
+
+    This archive is a tarfile that contains multiple compressed archives:
+
+    output.tar
+    |- version
+    |- dependencies.tar.gz
+    |- workspace.tar.gz
+    |- metadata.tar.gz
+
+    :param install_base: Directory with built artifacts from the workspace
+    :param staging_path: Directory where all dependencies have been installed
+    :param bundle_base: Directory to place the output of this function
+    :param metadata_paths: [str] paths to files which should be included
+    in the metadata archive
+    :param dependencies_changed: Boolean representing whether the staging path
+    needs to be re-archvied
+    """
+    logger.info('Archiving the bundle output')
+    print('Creating bundle archive V2...')
+
+    archive_tar_path = os.path.join(bundle_base, 'output.tar')
+    workspace_tar_gz_path = os.path.join(bundle_base, 'workspace.tar.gz')
+
+    # Install directory
+    workspace_staging_path = os.path.join(bundle_base, 'workspace_staging')
+    workspace_path = os.path.join(workspace_staging_path, 'workspace')
+    shutil.rmtree(workspace_staging_path, ignore_errors=True)
+    assets_directory = os.path.join(
+        os.path.dirname(os.path.realpath(__file__)), 'assets')
+
+    shellscript_path = os.path.join(assets_directory, 'workspace_setup.sh')
+    os.mkdir(workspace_staging_path)
+    shutil.copy2(shellscript_path,
+                 os.path.join(workspace_staging_path, 'setup.sh'))
+    shutil.copytree(install_base, workspace_path)
+    _recursive_tar_in_path(workspace_tar_gz_path, install_base,
```

Comment:
Should we have workspace_staging_path insted of install_base?

Reply:
Yep, there are a couple errors in this section, I think I've fixed them.